src/main/java/de/tum/in/www1/artemis/repository/ParticipationVCSAccessTokenRepository.java

```diff
@@ -0,0 +1,67 @@
+package de.tum.in.www1.artemis.repository;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.participation.ParticipationVCSAccessToken;
+import de.tum.in.www1.artemis.repository.base.ArtemisJpaRepository;
+
+@Profile(PROFILE_CORE)
+@Repository
+public interface ParticipationVCSAccessTokenRepository extends ArtemisJpaRepository<ParticipationVCSAccessToken, Long> {
+
+    /**
+     * Delete all participation vcs access token that belong to the given participation
+     *
+     * @param participationId the id of the participation where the tokens should be deleted
+     */
+    @Transactional // ok because of delete
+    @Modifying
+    void deleteByParticipation_id(long participationId);
```

Comment:
Why do we need `_id` here but the method below works with `Id`?
I'd say it would be worth it to write a query statement, so we can use proper method naming

Reply:
Is this some kind of Springboot black magic asking for that kind of naming in order to resolve the functionality automatically? 😆

Reply:
I think functionality wise it does not make a difference (chatgpt says it makes a slight difference in how JPA generates the query, but I'm not sure if that should be believed).
Seems like in older parts of the code the ....._Id(...) notation is still used in some places.
Here the notation without underscore works as well, so I replaced it :+1: